RFS/src/main/java/org/opensearch/migrations/bulkload/worker/DocumentsRunner.java

```diff
@@ -57,14 +85,50 @@ public CompletionStatus migrateNextShard(
                 }
             }, new IWorkCoordinator.WorkAcquisitionOutcomeVisitor<>() {
                 @Override
-                public CompletionStatus onAlreadyCompleted() throws IOException {
+                public CompletionStatus onAlreadyCompleted() {
                     return CompletionStatus.NOTHING_DONE;
                 }
 
                 @Override
                 public CompletionStatus onAcquiredWork(IWorkCoordinator.WorkItemAndDuration workItem) {
-                    doDocumentsMigration(IndexAndShardCursor.valueFromWorkItemString(workItem.getWorkItemId()), context);
-                    return CompletionStatus.WORK_COMPLETED;
+                    var docMigrationCursors = setupDocMigration(workItem.getWorkItem(), context);
+                    var latch = new CountDownLatch(1);
+                    var finishScheduler = Schedulers.newSingle( "workFinishScheduler");
+                    var disposable = docMigrationCursors
```

Comment:
Why is this not its own cancel/dispose (or autocloseable) method?  Are you trying to make sure that only the creator has access to cancel the class via this back-door hook?  That doesn't seem like it needs to be a critical concern and it makes the code much harder to maintain